modules/imgproc/src/smooth.cpp

```diff
@@ -3326,57 +3374,59 @@ template <>
-        v_src10 = v_load((int16_t*)(src[1]) + i);
-        v_src11 = v_load((int16_t*)(src[1]) + i + 8);
-        v_src12 = v_load((int16_t*)(src[1]) + i + 16);
-        v_src13 = v_load((int16_t*)(src[1]) + i + 24);
+        v_src00 = vx_load((int16_t*)(src[0]) + i);
+        v_src01 = vx_load((int16_t*)(src[0]) + i + VECSZ);
+        v_src02 = vx_load((int16_t*)(src[0]) + i + 2*VECSZ);
+        v_src03 = vx_load((int16_t*)(src[0]) + i + 3*VECSZ);
+        v_src10 = vx_load((int16_t*)(src[1]) + i);
+        v_src11 = vx_load((int16_t*)(src[1]) + i + VECSZ);
+        v_src12 = vx_load((int16_t*)(src[1]) + i + 2*VECSZ);
+        v_src13 = vx_load((int16_t*)(src[1]) + i + 3*VECSZ);
         v_zip(v_add_wrap(v_src00, v_128), v_add_wrap(v_src10, v_128), v_tmp0, v_tmp1);
-        v_int32x4 v_res0 = v_dotprod(v_tmp0, v_mul);
-        v_int32x4 v_res1 = v_dotprod(v_tmp1, v_mul);
+        v_int32 v_res0 = v_dotprod(v_tmp0, v_mul);
+        v_int32 v_res1 = v_dotprod(v_tmp1, v_mul);
         v_zip(v_add_wrap(v_src01, v_128), v_add_wrap(v_src11, v_128), v_tmp0, v_tmp1);
-        v_int32x4 v_res2 = v_dotprod(v_tmp0, v_mul);
-        v_int32x4 v_res3 = v_dotprod(v_tmp1, v_mul);
+        v_int32 v_res2 = v_dotprod(v_tmp0, v_mul);
+        v_int32 v_res3 = v_dotprod(v_tmp1, v_mul);
         v_zip(v_add_wrap(v_src02, v_128), v_add_wrap(v_src12, v_128), v_tmp0, v_tmp1);
-        v_int32x4 v_res4 = v_dotprod(v_tmp0, v_mul);
-        v_int32x4 v_res5 = v_dotprod(v_tmp1, v_mul);
+        v_int32 v_res4 = v_dotprod(v_tmp0, v_mul);
+        v_int32 v_res5 = v_dotprod(v_tmp1, v_mul);
         v_zip(v_add_wrap(v_src03, v_128), v_add_wrap(v_src13, v_128), v_tmp0, v_tmp1);
-        v_int32x4 v_res6 = v_dotprod(v_tmp0, v_mul);
-        v_int32x4 v_res7 = v_dotprod(v_tmp1, v_mul);
+        v_int32 v_res6 = v_dotprod(v_tmp0, v_mul);
+        v_int32 v_res7 = v_dotprod(v_tmp1, v_mul);
 
         int j = 2;
         for (; j < n - 1; j+=2)
         {
-            v_mul = v_reinterpret_as_s16(v_setall_u32(*((uint32_t*)(m+j))));
+            v_mul = v_reinterpret_as_s16(vx_setall_u32(*((uint32_t*)(m+j))));
 
-            v_src00 = v_load((int16_t*)(src[j]) + i);
-            v_src01 = v_load((int16_t*)(src[j]) + i + 8);
-            v_src02 = v_load((int16_t*)(src[j]) + i + 16);
-            v_src03 = v_load((int16_t*)(src[j]) + i + 24);
-            v_src10 = v_load((int16_t*)(src[j+1]) + i);
-            v_src11 = v_load((int16_t*)(src[j+1]) + i + 8);
-            v_src12 = v_load((int16_t*)(src[j+1]) + i + 16);
-            v_src13 = v_load((int16_t*)(src[j+1]) + i + 24);
+            v_src00 = vx_load((int16_t*)(src[j]) + i);
+            v_src01 = vx_load((int16_t*)(src[j]) + i + VECSZ);
+            v_src02 = vx_load((int16_t*)(src[j]) + i + 2*VECSZ);
+            v_src03 = vx_load((int16_t*)(src[j]) + i + 3*VECSZ);
+            v_src10 = vx_load((int16_t*)(src[j+1]) + i);
+            v_src11 = vx_load((int16_t*)(src[j+1]) + i + VECSZ);
```

Comment:
May be avoid redundant code to improve readability (compiler is able to optimize this in anyway):
```
const int16_t* pSrc0 = (const int16_t*)src[j] + i;
const int16_t* pSrc1 = (const int16_t*)src[j + 1] + i;
v_src00 = vx_load(pSrc0);
v_src01 = vx_load(pSrc0 + VECSZ);
v_src02 = vx_load(pSrc0 + 2*VECSZ);
v_src03 = vx_load(pSrc0 + i + 3*VECSZ);
v_src10 = vx_load(pSrc1);
v_src11 = vx_load(pSrc1 + VECSZ);
v_src12 = vx_load(pSrc1 + 2*VECSZ);
v_src13 = vx_load(pSrc1 + i + 3*VECSZ);
```